#! /bin/sh /usr/share/dpatch/dpatch-run
## 01_updatedb-withnew-su.dpatch by  Chronos Tachyon
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: bug #366269

@DPATCH@
diff -urNad findutils-4.2.27.orig/locate/updatedb.sh /tmp/dpep.t05W2A/findutils-4.2.27.orig/locate/updatedb.sh
--- findutils-4.2.27.orig/locate/updatedb.sh	2005-06-19 11:13:14.000000000 +0200
+++ /tmp/dpep.t05W2A/findutils-4.2.27.orig/locate/updatedb.sh	2006-05-13 13:46:10.303199192 +0200
@@ -84,11 +84,11 @@
 
 # figure out if su supports the -s option
 select_shell() {
-    if su "$1" -s $SHELL false < /dev/null  ; then
+    if su "$1" -s $SHELL -c false < /dev/null  ; then
 	# No.
 	echo ""
     else
-	if su "$1" -s $SHELL true < /dev/null  ; then
+	if su "$1" -s $SHELL -c true < /dev/null  ; then
 	    # Yes.
 	    echo "-s $SHELL"
         else
